### PR TITLE
add NEWS.md entry for deprecation of Array(T, dims...)

### PR DESCRIPTION
**Pull request by Sacha0**
Ref. #21475. Best!

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -439,6 +439,11 @@ Deprecated or removed
 
   * `cummin` and `cummax` have been deprecated in favor of `accumulate`.
 
+  * The idiosyncratic `Array` constructor syntax `Array(T, dims...)` has been deprecated
```
Comment (fredrikekre) [inline NEWS.md]: > idiosyncratic

I am learning so many new English words reading your PR's :)

Comment (tkelman) [inline NEWS.md]: the old syntax does match how you would call `zeros` or `ones` ... though those aren't constructors

Comment (Sacha0) [inline NEWS.md]: Would you prefer different wording? For example "The `zeros(T, dims...)`/`ones(T, dims...)`-like `Array` constructor syntax `Array(T, dims...)`..."? Or something else? Best! :)

Comment (tkelman) [inline NEWS.md]: I think it's okay, I just wouldn't have used the word idiosyncratic here

Comment (Sacha0) [inline NEWS.md]: "idiosyncratic" and "more consistent" removed :). Thanks!

**Pull request merged**